R/utils_plotting.R

```diff
@@ -253,15 +253,20 @@ NULL
 
 # Adjusted function originally developed for scater package by Aaron Lun
 .get_point_args <- function(colour_by, shape_by, size_by, alpha = 0.65,
-                            size = NULL) 
+                            size = NULL, shape = 21) 
 {
     aes_args <- list()
     fill_colour <- TRUE
     if (!is.null(shape_by)) {
         aes_args$shape <- "shape_by"
     }
     if (!is.null(colour_by)) {
-        aes_args$fill <- "colour_by"
+        # Only shapes 21 to 25 can be filled. Filling does not work in other shapes.
+        if(shape >= 21 && shape <= 25){
+            aes_args$fill <- "colour_by"
+        } else {
+            aes_args$colour <- "colour_by"
+        }
```

Comment:
If you code it like this you also have to update the value of `fill_colour`.

Reply:
Added